internal/services/openai.go

```diff
@@ -69,12 +73,25 @@ func (o openAI) askChatGPT(body io.Reader) (*ChatGPTResponse, error) {
 	req.Header.Set("Content-Type", "application/json")
 	req.Header.Set("Authorization", "Bearer "+o.token)
 
+	var currentReqResponseTime time.Duration
+	trace := &httptrace.ClientTrace{
+		GotFirstResponseByte: func() {
+			currentReqResponseTime = time.Since(time.Now())
```

Comment:
`time.Since(time.Now())` doesn't seem right. That should be very close to zero. Maybe a variable in the enclosing function that holds the start time?